### PR TITLE
[Payment]: add PaymentController with the corresponding logic 

### PR DESCRIPTION
**Pull request by MinoDentori**
**Added:**

- Stripe dependency
- PaymentController
- PaymentService with the implementation
- PaymentRepository
- SessionService with the implementation
- UriService with the implementation
- CreatePaymentDto
- UpdatePaymentDto

Changed:
- Updated Payment model, added new Status

### CODE REVIEW

File: src/main/java/com/example/accommodationbookingapp/service/date/impl/DateServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+public class DateServiceImpl implements DateService {
+    public static final int GAP = 1;
+
+    @Override
+    public long calculateDurationInDays(LocalDate checkInDate, LocalDate checkOutDate) {
+        Duration duration = Duration.between(
+                checkInDate.atStartOfDay(), checkOutDate.atStartOfDay());
+        return duration.toDays() + GAP;
+    }
+}
```
Comment (TarasYurkovskyi) [inline src/main/java/com/example/accommodationbookingapp/service/date/impl/DateServiceImpl.java:16]: @Service

**Review by TarasYurkovskyi — APPROVED** [review]
Good job!

File: src/main/java/com/example/accommodationbookingapp/controller/PaymentController.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    private PaymentService paymentService;
+    private UriService uriService;
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping("/{id}")
+    @Operation(summary = "Get Payments by User ID",
+            description = "Get Payments as dto by User ID")
+    public List<PaymentResponseDto> getPaymentsByUserId(@PathVariable Long id) {
+        return paymentService.findAllByUserId(id);
+    }
```
Comment (vadymhrnk) [inline src/main/java/com/example/accommodationbookingapp/controller/PaymentController.java:38]: I don't know if I understand this correctly but wouldn't it be better to send user id based on Authentication principal?

File: src/main/java/com/example/accommodationbookingapp/service/payment/PaymentService.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.example.accommodationbookingapp.service.payment;
+
+import com.example.accommodationbookingapp.dto.payment.PaymentResponseDto;
+import com.example.accommodationbookingapp.model.Payment;
+import java.net.MalformedURLException;
+import java.util.List;
+
+public interface PaymentService {
+
```
Comment (vadymhrnk) [inline src/main/java/com/example/accommodationbookingapp/service/payment/PaymentService.java:9]: Redundant empty line

File: src/main/java/com/example/accommodationbookingapp/service/payment/PaymentService.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    List<PaymentResponseDto> findAllByUserId(Long userId);
+
+    PaymentResponseDto create(Long bookingId) throws MalformedURLException;
+
+    PaymentResponseDto update(String sessionId, Payment.Status status);
+}
+
```
Comment (vadymhrnk) [inline src/main/java/com/example/accommodationbookingapp/service/payment/PaymentService.java:16]: same here

**Review by vadymhrnk — APPROVED** [review]
Good job! Everything looks good, left some suggestions for you to consider.

File: src/main/java/com/example/accommodationbookingapp/controller/PaymentController.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    public PaymentResponseDto initiatePaymentSession(@RequestBody CreatePaymentDto requestDto,
+                                                     HttpServletResponse response)
+            throws IOException {
+        PaymentResponseDto createdPaymentDto = paymentService.create(requestDto.getBookingId());
+        URI redirectUrl = uriService.buildUriWithSessionId(
+                createdPaymentDto.getSessionId(),
+                createdPaymentDto.getSessionUrl());
+        response.sendRedirect(String.valueOf(redirectUrl));
+        return createdPaymentDto;
+
```
Comment (DPotapov93) [inline src/main/java/com/example/accommodationbookingapp/controller/PaymentController.java:53]: Redundant empty line

**Review by DPotapov93 — APPROVED** [review]
Great Job!

**Pull request merged**